packages/@lwc/template-compiler/src/shared/parse5.ts

```diff
@@ -1,25 +1,44 @@
 /*
- * Copyright (c) 2018, salesforce.com, inc.
+ * Copyright (c) 2023, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import * as parse5 from 'parse5';
+import { Token } from 'parse5';
+import type { DefaultTreeAdapterMap } from 'parse5';
+import type { Tokenizer } from 'parse5';
 
-export function isElementNode(node: parse5.Node): node is parse5.Element {
-    return 'tagName' in node;
+export type Preprocessor = Tokenizer['preprocessor'];
+export type ElementLocation = Token.ElementLocation;
+export type ChildNode = DefaultTreeAdapterMap['childNode'];
+export type CommentNode = DefaultTreeAdapterMap['commentNode'];
+export type Document = DefaultTreeAdapterMap['document'];
+export type DocumentFragment = DefaultTreeAdapterMap['documentFragment'];
+export type Element = DefaultTreeAdapterMap['element'];
+export type Node = DefaultTreeAdapterMap['node'];
+export type Template = DefaultTreeAdapterMap['template'];
+export type TextNode = DefaultTreeAdapterMap['textNode'];
+
+export { Token };
+
+export function isElementNode(node: Node | null): node is Element {
+    return node !== null && 'tagName' in node;
 }
 
-export function isCommentNode(node: parse5.Node): node is parse5.CommentNode {
+export function isCommentNode(node: Node): node is CommentNode {
     return node.nodeName === '#comment';
 }
 
-export function isTextNode(node: parse5.Node): node is parse5.TextNode {
+export function isTextNode(node: Node): node is TextNode {
     return node.nodeName === '#text';
 }
 
+export function isTemplateNode(node: Node): node is Template {
+    return isElementNode(node) && node.tagName === 'template';
+}
+
 export function getTemplateContent(
-    templateElement: parse5.Element
-): parse5.DocumentFragment | undefined {
-    return (templateElement as any).content;
+    templateElement: Element | Template
+): DocumentFragment | undefined {
+    return isTemplateNode(templateElement) ? templateElement.content : undefined;
```

Comment:
Consider updating the type signature to only accept template nodes, and update the call sites accordingly.